tests/MongoDB.Driver.Tests/Specifications/unified-test-format/EntityMap.cs

```diff
@@ -0,0 +1,478 @@
+
+            return (client, eventCapturer);
+        }
+
+        private IMongoCollection<BsonDocument> CreateCollection(BsonDocument entity)
+        {
+            string collectionName = null;
+            IMongoDatabase database = null;
+            MongoCollectionSettings settings = null;
+
+            foreach (var element in entity)
+            {
+                switch (element.Name)
+                {
+                    case "id":
+                        // handled on higher level
+                        break;
+                    case "database":
+                        var databaseId = entity["database"].AsString;
+                        database = _databases[databaseId];
+                        break;
+                    case "collectionName":
+                        collectionName = entity["collectionName"].AsString;
+                        break;
+                    case "collectionOptions":
+                        settings = new MongoCollectionSettings();
+                        foreach (var option in element.Value.AsBsonDocument)
+                        {
+                            switch (option.Name)
+                            {
+                                case "readConcern":
+                                    settings.ReadConcern = ReadConcern.FromBsonDocument(option.Value.AsBsonDocument);
+                                    break;
+                                default:
+                                    throw new FormatException($"Unrecognized collection option field: '{option.Name}'");
+                            }
+                        }
+                        break;
+                    default:
+                        throw new FormatException($"Unrecognized collection entity field: '{element.Name}'");
+                }
+            }
+
+            return database.GetCollection<BsonDocument>(collectionName, settings);
+        }
+
+        private IMongoDatabase CreateDatabase(BsonDocument entity)
+        {
+            IMongoClient client = null;
+            string databaseName = null;
+
+            foreach (var element in entity)
+            {
+                switch (element.Name)
+                {
+                    case "id":
+                        // handled on higher level
+                        break;
+                    case "client":
+                        var clientId = element.Value.AsString;
+                        client = _clients[clientId];
+                        break;
+                    case "databaseName":
+                        databaseName = element.Value.AsString;
+                        break;
+                    default:
+                        throw new FormatException($"Unrecognized database entity field: '{element.Name}'");
+                }
+            }
+
+            return client.GetDatabase(databaseName);
+        }
+
+        private (IClientSessionHandle, BsonDocument) CreateSession(BsonDocument entity)
```

Comment:
Change to:

    private void CreateSession(BsonDocument entity, out IClientSessionHandle session, out BsonDocument sessionId)

to avoid annoying `Item1` and `Item2`.

For example... neither then name `Item2` or the type `BsonDocument` contain the slightest hint that the value is a `sessionId` (which a named out parameter does).

Reply:
Also: consider that the returned `session` object contains the `sessionId` inside of it so it might not even be necessary to return two values here. Maybe just returning the `session` is sufficient?